code/modules/vehicles/mecha/equipment/weapons/weapons.dm

```diff
@@ -596,11 +600,12 @@
 			carbontarget.update_handcuffed()
 		return
 
-	if(!istype(target, /obj/machinery/door))
+	if(istype(target, /obj/machinery/door))
+		. = ..()
```

Comment:
man doing the call to parent like this is really disgusting but i don't see a better way of doing it unless you wanna do something like

```dm
if(!action_checks(target))
 return

if(!isliving(target) && !istype(target, doortype))
 return

. = ..()

if(isliving(target))
 paddy_claw_living_proc()
 return

if(isdoor(target)) // yes i know this doesn't exist just illustrative
 padd_claw_door_proc()
 return
```

feels a bit cleaner rather than having a huge chunk of fuck off text as well as prevent (what I find to be) the really weird practice of having the parent call be conditional in cases that aren't early returns.

Reply:
It is a little ugly but I don't think this is any *better* IMO